test/framework/gardener_utils.go

```diff
@@ -576,3 +596,142 @@ func ScaleGardenerScheduler(setupContextTimeout time.Duration, client client.Cli
 	return ScaleDeployment(setupContextTimeout, client, desiredReplicas, "gardener-controller-manager", gardencorev1beta1constants.GardenNamespace)
 }
+
+// CreateShoot Creates a shoot from a seed Object and waits until it is successfully reconciled
+func (f *GardenerFramework) CreateSeed(ctx context.Context, seed *gardencorev1beta1.Seed) error {
+	err := retry.UntilTimeout(ctx, 20*time.Second, 5*time.Minute, func(ctx context.Context) (done bool, err error) {
+		_, err = f.createSeedResource(ctx, seed)
+		if apierrors.IsInvalid(err) || apierrors.IsForbidden(err) || apierrors.IsAlreadyExists(err) {
+			return retry.SevereError(err)
+		}
+		if err != nil {
+			f.Logger.Debugf("unable to create seed %s: %s", seed.Name, err.Error())
+			return retry.MinorError(err)
+		}
+		return retry.Ok()
+	})
+	if err != nil {
+		return err
+	}
+
+	// Then we wait for the shoot to be created
+	err = f.WaitForSeedToBeCreated(ctx, seed)
+	if err != nil {
+		return err
+	}
+
+	f.Logger.Infof("Seed %s was created!", seed.Name)
+	return nil
+}
+
+// createShootResource creates a shoot from a shoot Object
+func (f *GardenerFramework) createSeedResource(ctx context.Context, seed *gardencorev1beta1.Seed) (*gardencorev1beta1.Seed, error) {
+	err := f.checkIfSeedExists(ctx, seed)
+	if err == nil {
+		return seed, apierrors.NewAlreadyExists(gardencorev1beta1.Resource("seed"), seed.Name)
+	}
+	if !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	if err := f.GardenClient.DirectClient().Create(ctx, seed); err != nil {
+		f.Logger.Errorf("Failed to create SEED: %v ", err)
+		return nil, err
+	}
+	f.Logger.Infof("Seed resource %s was created!", seed.Name)
+	return seed, nil
+}
+
+func (f *GardenerFramework) checkIfSeedExists(ctx context.Context, seed *gardencorev1beta1.Seed) error {
+	return f.GardenClient.DirectClient().Get(ctx, client.ObjectKey{Name: seed.Name}, seed)
+}
+
+// WaitForShootToBeCreated waits for the shoot to be created
+func (f *GardenerFramework) WaitForSeedToBeCreated(ctx context.Context, seed *gardencorev1beta1.Seed) error {
+	return retry.UntilTimeout(ctx, 30*time.Second, 60*time.Minute, func(ctx context.Context) (done bool, err error) {
+		newSeed := &gardencorev1beta1.Seed{}
+		err = f.GardenClient.DirectClient().Get(ctx, client.ObjectKey{Name: seed.Name}, newSeed)
+		if err != nil {
+			f.Logger.Infof("Error while waiting for seed to be created: %s", err.Error())
+			return retry.MinorError(err)
+		}
+		*seed = *newSeed
+		completed, msg := SeedRegistrationCompleted(seed)
+		if completed {
+			return retry.Ok()
+		}
+		f.Logger.Infof("Seed %s not yet created successfully (%s)", seed.Name, msg)
+
+		return retry.MinorError(fmt.Errorf("seed %q was not successfully reconciled", seed.Name))
+	})
+}
+
+// SeedreationCompleted checks if a seed is successfully reconciled. In case it is not, it also returns a descriptive message stating the reason.
+func SeedRegistrationCompleted(newSeed *gardencorev1beta1.Seed) (bool, string) {
```

Comment:
seems like an internal func

Reply:
Can we use this https://github.com/gardener/gardener/blob/master/pkg/utils/kubernetes/health/health.go#L244-L264